src/nix/modules/service/docker-compose-service.nix

```diff
@@ -101,10 +101,33 @@ in
       description = dockerComposeRef "container_name";
     };
     service.depends_on = mkOption {
-      type = listOf str;
+      type = either (listOf str) (attrsOf (attrsOf str));
```

Comment:
This doesn't seem accurate. I think it should be something like

```suggestion
      type = either (listOf str) (attrsOf (submodule someModule));
```

Where `someModule` is a module that declares the `condition` option.

Reply:
Done!